test/index.js

```diff
@@ -303,14 +294,13 @@ describe('getService()', () => {
       const consulite = new Consulite({ consul: `http://localhost:${server.address().port}` });
       consulite.getService('notfound', (err, service) => {
         expect(err).to.exist();
-        done();
       });
     });
   });
 });
 
 describe('getServiceHosts()', () => {
-  it('returns service hosts from consul', (done) => {
+  it('returns service hosts from consul', () => {
```

Comment:
The tests that are async will need to be marked as async and will need to be awaited correctly. It may be helpful to use [cb-barrier](http://npmjs.com/cb-barrier) to block on the listen events. For example:

```js
it('returns service hosts from consul', async () => {
    const barrier = new Barrier();
    const server = Http.createServer((req, res) => {
      res.writeHead(200, { 'Content-Type': 'application/json' });
      res.end(JSON.stringify([
        { Service: { Address: 'foo1.com', Port: '1234' } },
        { Service: { Address: 'foo2.com', Port: '1234' } }
      ]));
    });

    server.on('error', (err) => {
      barrier.pass(err);
    });

    server.listen(0, () => {
      const consulite = new Consulite({ consul: `http://localhost:${server.address().port}` });

      consulite.getServiceHosts('foobar', (err, hosts) => {
        expect(err).to.not.exist();
        expect(hosts.length).to.equal(2);
        expect(hosts[0].address).to.equal('foo1.com');
        expect(hosts[1].address).to.equal('foo2.com');
        barrier.pass();
      });
    });

    await barrier;
  });
```

Reply:
Thank you for your comments! When I use barrier in the way you suggested, I get `Timed out (2000ms)` errors when running `npm run test` - do you see those, too? Maybe I misunderstood how to correctly use `cb-barrier`?

Reply:
Ya, I think the test will need to be updated if the callback style is no longer used for `consulite.getServiceHosts`. There may be other tests that also need to be updated to `await consulite.getServiceHosts`

Reply:
Okay so if I understand it correctly all async functions in `lib/index.js` should get their async tests with proper async / await sugar? Are you okay with dropping the callback style like that?

Reply:
I have a first draft of async/await tests in e5cee3d, I'm curious about what you think about it.

Reply:
Is there anything that can be done to move this along?